docs/limitations.rst

```diff
@@ -92,100 +91,168 @@ the "sync-barrier-sync" construct needs to be used.
 .. Note::
 
     The "barrier" in "sync-barrier-sync" can be replaced by a send-recv or
-    certain collectives that are guaranteed to be synchronized. See the "Note on
-    the third step" in the `VerifyIO README`_ for more information.
+    certain collectives that are guaranteed to be synchronized.
+    See the "Note on the third step" in the `VerifyIO README`_
+    for more information.
 
-**Cost:** If making edits to the application source code is an option, the
-amount of time and effort required to track down all the places that proper
-synchonization calls are needed can be very labor intensive.
-:doc:`VerifyIO <verifyio>` can help with in this effort.
+Proper data consistency synchronization is also required
+between MPI-I/O calls that imply write/read operations.
+For example, ``MPI_File_set_size()`` and ``MPI_File_preallocate()``
+act as write operations,
+and ``MPI_File_get_size()`` acts as a read operation.
+There may be other MPI-I/O calls that imply write/read operations.
 
-HDF5 FILE_SYNC
-""""""""""""""
+Manually Add Sync Operations
+""""""""""""""""""""""""""""
 
-HDF5 provided config option that forces HDF5 to add an ``MPI_File_sync()`` call
-after every collective write operation when needed by the underlying MPI-IO
-driver. Set ``HDF5_DO_MPI_FILE_SYNC=1`` to enable this option.
+Data consistency in UnifyFS is designed to be compatible
+with MPI-I/O application-managed file consistency semantics.
+An application that follows proper MPI-I/O file consistency
+semantics should run correctly on UnifyFS,
+provided that the ``MPI_File_sync()`` implementation flushes
+newly written data to UnifyFS.
 
-.. Note::
+On POSIX-compliant parallel file systems like Lustre,
+many applications may run correctly
+even when they are missing sufficient file consistency synchronization.
+To run correctly on UnifyFS, an application must make
+all ``MPI_File_sync()`` calls required by the MPI standard.
 
-    This option will soon be available in the `HDF5 develop branch`_ as well as
-    in the next HDF5 release.
+**Cost:** It may be labor intensive to identify and correct all places
+within an application where file synchronization calls are required.
+The :doc:`VerifyIO <verifyio>` tool can assist with this effort.
 
-**Cost:** Can cause a significant decrease in write performance as the amount of
-file sync operations performed will likely be more than necessary. Similar to,
-but potentially more efficient than, the ``WRITE_SYNC`` workaround as less
-overall file syncs may be performed in comparision, but still likely more than
-needed.
+.. TODO: Mention use/need of ``romio_visibility_immediate`` hint once available.
+.. https://github.com/pmodels/mpich/issues/5902
```

Comment:
This also appears to have been completed and merged into MPICH in Sept of 2022 (https://github.com/pmodels/mpich/pull/6148).

Looks like this will be available in the v4.1 release as it's currently in 4.1rc1 (https://github.com/pmodels/mpich/blob/5fa0cc78e256d44de2d01d63e90167e2de1956b7/src/mpi/romio/adio/include/adioi.h#L51)

Is my understanding correctly that this is the romio hint HDF5 is waiting for (for which they created `HDF5_DO_MPI_FILE_SYNC` as a workaround)?
https://github.com/HDFGroup/hdf5/pull/1801